--- ./src/zforth/src/zforth/zforth.h	2020-08-02 11:45:12.234846683 -0700
+++ ./src/zforth/src/zforth/zforth.h	2020-08-02 11:45:50.057740588 -0700
@@ -1,6 +1,8 @@
 #ifndef zforth_h
 #define zforth_h
 
+#include <stdarg.h>
+
 #include "zfconf.h"
 
 /* Abort reasons */
--- ./src/zforth/src/zforth/zforth.c.orig	2020-02-25 03:32:57.000000000 -0800
+++ ./src/zforth/src/zforth/zforth.c	2020-08-05 18:57:36.975689160 -0700
@@ -142,8 +142,13 @@
 }
 
 #else
-static void trace(const char *fmt, ...) { }
-static const char *op_name(zf_addr addr) { return NULL; }
+static void trace(const char *fmt, ...) {
+	(void)fmt;
+}
+static const char *op_name(zf_addr addr) {
+	(void)addr;
+	return NULL;
+}
 #endif
 
 
@@ -883,13 +888,15 @@
 {
 	zf_result r = (zf_result)setjmp(jmpbuf);
 
+	const char *tmp = buf;
+
 	if(r == ZF_OK) {
 		for(;;) {
-			handle_char(*buf);
-			if(*buf == '\0') {
+			handle_char(*tmp);
+			if(*tmp == '\0') {
 				return ZF_OK;
 			}
-			buf ++;
+			tmp ++;
 		}
 	} else {
 		COMPILING = 0;
